grpc-server/workflow.go

```diff
@@ -181,6 +174,7 @@ func (s *server) ListWorkflows(_ *workflow.Empty, stream workflow.WorkflowServic
 			Hardware:  w.Hardware,
 			CreatedAt: w.CreatedAt,
 			UpdatedAt: w.UpdatedAt,
+			State:     getWorkflowState(s.db, stream.Context(), w.ID),
```

Comment:
you're making 2 calls to `getWorkflowState`, this means 2 DB calls. You could call `getWorkflowState` just once (1 DB call) and store it to a variable for use in both locations.

Reply:
I don't think it is possible to call `getWorkflowState` once and store it because the `workflow state` will change during the execution of a workflow which makes it dynamic. Therefore we have to fetch the `latest` value of a `workflow state`.

Reply:
Ah, apologies. You are right. For some reason I thought I saw two calls in a single function.